client/beadledom-client-test/src/test/scala/com/cerner/beadledom/client/ClientServiceSpec.scala

```diff
@@ -132,6 +132,17 @@ class ClientServiceSpec(contextRoot: String, servicePort: Int)
         results mustNot be(null)
         results.getStatus mustBe 400
       }
+
+      it("rejects 0 for limit") {
+        val injector = getInjector(List(new ResourceOneModule))
+
+        val paginatedResource = injector.getInstance(classOf[PaginatedClientResource])
+
+        val results: GenericResponse[JsonOneOffsetPaginatedListDto] = paginatedResource.index(1L, 0)
+
+        results mustNot be(null)
+        results.getStatus mustBe 400
```

Comment:
I've seen instances where people set the limit to 0 when they just want to get a total results count. Why not still have a 200 when limit is 0?

Reply:
https://github.com/cerner/beadledom/pull/118/commits/0fcf057b54008675463e241ffa2050c9d8855181 I have made the changed for it. With this, the result would be a single page with empty results block and totalResults would be the total number of results. And everything except firstLink will be null.  Something like this
```
{
  "items": [],
  "totalResults": 3,
  "firstLink": "http://<service-resource-URI>?offset=0&limit=0"
}
```
Also note that the offset will always be 0 since this is the first page (and the only page). Let me know if this is the acceptable behavior